### PR TITLE
Contribute: Event Bus Product API

### PR DESCRIPTION
**Pull request by noxer**
#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: api4/api.go [hunk 1]
```diff
@@ -139,6 +139,8 @@ type Routes struct {
 	InsightsForUser *mux.Router // 'api/v4/users/me/top'
 
 	Usage *mux.Router // 'api/v4/usage'
+
+	Events *mux.Router // 'api/v4/events'
```
Comment (isacikgoz) [inline api4/api.go:143]: I rather have this under the `system` path, would it make sense to you?

File: api4/events.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+func (api *API) InitEvents() {
+	// GET /api/v4/events/topics
+	api.BaseRoutes.Events.Handle("/topics", api.APISessionRequired(getTopics)).Methods("GET")
+	// GET /api/v4/events/topics/<topicId>
+	api.BaseRoutes.Events.Handle("/topics/{topic_id}", api.APISessionRequired(getTopic)).Methods("GET")
+	// GET /api/v4/events/schemas/<topicId>
+	api.BaseRoutes.Events.Handle("/schemas/{topic_id}", api.APISessionRequired(getSchema)).Methods("GET")
+}
+
+func getTopics(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (isacikgoz) [inline api4/events.go:25]: Can you add a permission check for only manage system scheme? Let's start this for admins only. Do yo have any thoughts @oh6hay?

File: api4/events.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+	// parameter withSchema is to be parsed
+	// parameter topicId
+	// parameter withSchema is to be parsed
+	withSchema, _ := strconv.ParseBool(r.FormValue("withSchema"))
+	topicId := c.Params.TopicId
+
+	topics, err := c.App.EventBroker().EventTypes()
+	if err != nil {
+		c.Err = model.NewAppError("getTopics", "app.broker_error", nil, "", http.StatusInternalServerError).Wrap(err)
```
Comment (isacikgoz) [inline api4/events.go:62]: Shall we create the translation strings as well?

File: api4/events.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	if err := json.NewEncoder(w).Encode(topic); err != nil {
+		c.Err = model.NewAppError("getTopics", "api.marshal_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func getSchema(c *Context, w http.ResponseWriter, r *http.Request) {
+	// calls broker to get a list of topics
+	// parameter topicId
+	topicId := c.Params.TopicId
```
Comment (isacikgoz) [inline api4/events.go:82]: Can we read this as required field?

File: api4/events.go (hunk 3 above)
Comment (noxer) [reply api4/events.go:62]: That was more of a placeholder.

File: api4/api.go (hunk 1 above)
Comment (noxer) [reply api4/api.go:143]: yes, sounds reasonable

File: api4/system.go [hunk 5]
```diff
@@ -71,6 +72,13 @@ func (api *API) InitSystem() {
 	api.BaseRoutes.System.Handle("/onboarding/complete", api.APISessionRequired(getOnboarding)).Methods("GET")
 	api.BaseRoutes.System.Handle("/onboarding/complete", api.APISessionRequired(completeOnboarding)).Methods("POST")
 	api.BaseRoutes.System.Handle("/schema/version", api.APISessionRequired(getAppliedSchemaMigrations)).Methods("GET")
+
+	// GET /api/v4/events/topics
```
Comment (isacikgoz) [inline api4/system.go:76]:
```suggestion
	// GET /api/v4/system/events/topics
```

This applies to similar occasions as well.

File: api4/system.go [hunk 6]
```diff
@@ -1008,3 +1016,88 @@ func checkHasNilFields(value any) bool {
 
 	return false
 }
+
+func getEventsTopics(c *Context, w http.ResponseWriter, r *http.Request) {
+	// parameter withSchema is to be parsed
+	// restrict only for admins
+	if !c.IsSystemAdmin() {
+		c.Err = model.NewAppError("getEventsTopics", "api.get_events_topics.permission_error", nil, "", http.StatusBadRequest)
```
Comment (isacikgoz) [inline api4/system.go:1024]:
```suggestion
		c.Err = model.NewAppError("getEventsTopics", "api.get_events_topics.permission_error", nil, "", http.StatusForbidden)
```

This applies to similar occasions as well.

**Review by isacikgoz — APPROVED** [review]
Looks good, just some styling issues needs to be addressed 👍

**Review by isacikgoz — APPROVED** [review]
LGTM, good to go after the build passes. 💯

**Pull request merged**